CHANGELOG/CHANGELOG-4.2.md

```diff
@@ -0,0 +1,243 @@
+# Release notes for v4.2.0
+
+[Documentation](https://kubernetes-csi.github.io)
+
+# Changelog since v4.1.0
+
+## Changes by Kind
+
+### Feature
+
+#### Snapshot APIs 
+
+- The namespace of the referenced VolumeSnapshot is printed when printing a VolumeSnapshotContent. ([#535](https://github.com/kubernetes-csi/external-snapshotter/pull/535), [@tsmetana](https://github.com/tsmetana))
+
+#### Snapshot Controller
+
+- `retry-interval-start` and `retry-interval-max` arguments are added to common-controller which controls retry interval of failed volume snapshot creation and deletion. These values set the ratelimiter for snapshot and content queues. ([#530](https://github.com/kubernetes-csi/external-snapshotter/pull/530), [@humblec](https://github.com/humblec))
+- Add command line arguments to configure leader election options for the snapshot controller. ([#575](https://github.com/kubernetes-csi/external-snapshotter/pull/575), [@bertinatto](https://github.com/bertinatto))
+- Adds an in-operations_in_flight metric for determining the number of snapshot operations in progress. ([#519](https://github.com/kubernetes-csi/external-snapshotter/pull/519), [@ggriffiths](https://github.com/ggriffiths))
+- Introduced "SnapshotCreated" and "SnapshotReady" events. ([#540](https://github.com/kubernetes-csi/external-snapshotter/pull/540), [@rexagod](https://github.com/rexagod))
+
+#### CSI Snapshotter Sidecar
+
+- `retry-interval-start` and `retry-interval-max` arguments are added to csi-snapshotter sidecar which controls retry interval of failed volume snapshot creation and deletion. These values set the ratelimiter for volumesnapshotcontent queue. ([#308](https://github.com/kubernetes-csi/external-snapshotter/pull/308), [@humblec](https://github.com/humblec))
+- Add command line arguments to configure leader election options for CSI snapshotter sidecar. ([#538](https://github.com/kubernetes-csi/external-snapshotter/pull/538), [@RaunakShah](https://github.com/RaunakShah))
+
+### Bug or Regression
+
+#### Snapshot Controller
+
+- Add process_start_time_seconds metric ([#569](https://github.com/kubernetes-csi/external-snapshotter/pull/569), [@saikat-royc](https://github.com/saikat-royc))
+- Adds the leader election health check for the snapshot controller at `/healthz/leader-election` ([#573](https://github.com/kubernetes-csi/external-snapshotter/pull/573), [@ggriffiths](https://github.com/ggriffiths))
+- Remove kube-system namespace verification during startup and instead list volumes across all namespaces ([#515](https://github.com/kubernetes-csi/external-snapshotter/pull/515), [@mauriciopoppe](https://github.com/mauriciopoppe))
+
+### Other (Cleanup or Flake)
```

Comment:
Is this worth a release note? https://github.com/kubernetes-csi/external-snapshotter/pull/550

Reply:
Sure.  Will add that.